instrumentation/opentelemetry-instrumentation-starlette/src/opentelemetry/instrumentation/starlette/__init__.py

```diff
@@ -184,18 +187,29 @@ def client_response_hook(span: Span, scope: dict[str, Any], message: dict[str, A
 from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
 from opentelemetry.instrumentation.starlette.package import _instruments
 from opentelemetry.instrumentation.starlette.version import __version__
-from opentelemetry.metrics import get_meter
+from opentelemetry.metrics import MeterProvider, get_meter
 from opentelemetry.semconv.trace import SpanAttributes
-from opentelemetry.trace import get_tracer
+from opentelemetry.trace import TracerProvider, get_tracer
 from opentelemetry.util.http import get_excluded_urls
 
+if TYPE_CHECKING:
+    from typing import NotRequired, TypedDict, Unpack
+
+    class InstrumentKwargs(TypedDict):
+        tracer_provider: NotRequired[TracerProvider]
+        meter_provider: NotRequired[MeterProvider]
```

Comment:
btw, there's a bug on this instrumentation. On the `_instrument` function, it's trying to retrieve the value from `_meter_provider` instead of `meter_provider`.

Reply:
Fixed in #3048